performance/runner/src/measure.rs

```diff
@@ -0,0 +1,89 @@
+use crate::exceptions::IOError;
+use std::fs;
+use std::path::PathBuf;
+use std::process::{Command, ExitStatus};
+
+// `Metric` defines a dbt command that we want to measure on both the
+// baseline and dev branches.
+#[derive(Debug, Clone)]
+struct Metric<'a> {
+    name: &'a str,
+    prepare: &'a str,
+    cmd: &'a str,
+}
+
+impl Metric<'_> {
+    // Returns the proper filename for the hyperfine output for this metric.
+    fn outfile(&self, project: &str, branch: &str) -> String {
+        [branch, "_", self.name, "_", project, ".json"].join("")
+    }
+}
+
+// Calls hyperfine via system command, and returns all the exit codes for each hyperfine run.
+pub fn measure<'a>(
+    projects_directory: &PathBuf,
+    dbt_branch: &str,
+) -> Result<Vec<ExitStatus>, IOError> {
+    /*
+        Strategy of this function body:
+        1. Read all directory names in `projects_directory`
+        2. Pair `n` projects with `m` metrics for a total of n*m pairs
+        3. Run hyperfine on each project-metric pair
+    */
+
+    // To add a new metric to the test suite, simply define it in this list:
+    // TODO: This could be read from a config file in a future version.
+    let metrics: Vec<Metric> = vec![Metric {
+        name: "parse",
+        prepare: "rm -rf target/",
+        cmd: "dbt parse --no-version-check",
```

Comment:
If there's a way to compare command-specific output, I'd be excited to add:
```
dbt --partial-parse parse
dbt --use-experimental-parser parse
```
For the moment, I don't think it's worth a lot of additional effort to add that in, since we plan on turning on both of these by default pretty soon (#3502).

Reply:
totally. Hyperfine can already run two things and compare them just like that. So to build that, I would have a third sub command that follows a different flow, but reuses pretty much all the same types so it wouldn't be hard to build at all if we wanted to track the performance differences. It wouldn't really be regression testing, but I think it's still valuable.